### ISSUE TITLE
Initialization including equality constraints / zero cones

### ISSUE THREAD

**Issue by mipals** [oxfordcontrol/Clarabel.jl#177]
I have a question regarding the initialization step with regards to the equality constraints / zero cone. In the preprint (in the case of $P\neq 0$, but a similar thing can be said in the case of $P=0$) it is stated that the following linear system is solved

$$
\begin{bmatrix}P & A^\top \\ 
A & -I\end{bmatrix}
    \begin{bmatrix}x \\ 
z\end{bmatrix}
    = \begin{bmatrix}-q \\ 
b\end{bmatrix}
$$

This would mean that the equality constraints are not necessarily satisfied (due to the presence of $I$). However, looking at the code the initialization seems to utilize the standard KKT-solver and even have a comment referring to the CVXOPT initialization 

https://github.com/oxfordcontrol/Clarabel.jl/blob/dd709aab1eeba5ff89f8f5813818ad6700cbd670/src/solver.jl#L365

Both cases point to towards that the equality constraints are actually satisfied. So I guess in practice you are solving something akin to

$$
\begin{bmatrix}P & A^\top \\ 
A & -I_E\end{bmatrix}
    \begin{bmatrix}x \\ 
z\end{bmatrix}
    = \begin{bmatrix}-q \\ 
b\end{bmatrix},
$$

where $I_E = I - E$ with $E$ being a matrix with ones on the diagonal entries corresponding to equality constraints?

**Comment by goulart-paul** [comment]
You are right, the solver is initialized in the symmetric cone case using (nearly) the second form, and not quite as in the paper.    Instead it is like CVXOPT, where they explicitly partition their constraint matrices into equality and inequality terms (see section 5.3 / equation (20) of the CVXOPT paper) and use a zero block on the diagonal part that goes with the equality term.

We achieve the same effect (without using separate equality/inequality constraint matrices) by instead allowing a different "identity scaling" for every cone type in the symmetric initialization.   The key line is here:

https://github.com/oxfordcontrol/Clarabel.rs/blob/7036899fda1c3c95bbee894bd9babce4f2e66c40/src/solver/core/solver.rs#L469

which calls the `set_identity_scaling` on each of the constituent cones to populate their bit of the lower right hand side.   For all cones other than the zero cone, we just use the identity matrix.   For the zero cone we leave it as a zero block instead.

I say 'nearly' above since we still apply static regularization so that the LHS will be quasidefinite.
